### ISSUE TITLE
Main menu appears sometimes at the wrong place on an external display when panel is auto-hide

### ISSUE THREAD

**Issue by ralf3u** [lxqt/lxqt#2414]
I use a 15.6'' notebook. It is connected to a 13.3'' eInk-display. The panel is in auto-hide. When closing a window and I want to let appear the main menu, then the main menu appears sometimes at the wrong place on an external display (see image below). When the main menu appears at the wrong place then I think it is always at the same position: on the left border of the screen, somewhere between top and bottom. I can't reproduce the issue step by step but sometimes it happens. If I disconnect the external display, then on the notebook display I can see that the main menu is at the correct place. But if I reconnect the notebook to the external display, then on the external display the main menu is still at the wrong place.
![main_menu_wrong_position](https://user-images.githubusercontent.com/68117017/234585263-f625ab3f-ca3c-4445-a200-33384ea0e9b2.jpg)

Possible problem
I can imagine that the issue has something to do with the panel that is in auto-hide. And I can imagine that it has to do something with the different sizes of the displays. I think that the not-correct-positioned main menu appears on the external display where it would appear on a 15,6'' display.

One more issue
I think that if the main menu is not at the right position, then the unfocused-desktops are also shown not correctly on the eInk-display because there is a black line at the bottom:
![main_menu_desktops](https://user-images.githubusercontent.com/68117017/234585782-52269502-e81f-457b-acfb-47ff133be109.jpg)

Lubuntu 23.04 LXQt 1.2.0

**Comment by tsujan** [comment]
Which WM do you use? If its Openbox, it's known to have bugs like this.

**Comment by ralf3u** [comment]
>If it's Openbox, it's known to have bugs like this.

I can imagine that you are right when it comes to the first picture.

But I think that the second picture is not related to Openbox because when I select different LXQt Theme, then the artifacts are different. I intend to open a new issue for that the next days.

**Comment by tsujan** [comment]
> I intend to open a new issue for that the next days.

Please don't. LXQt is in no way responsible for rendering anything.

**Comment by ralf3u** [comment]
>Please don't. LXQt is in no way responsible for rendering anything.

The different artifacts with black bars are only visible on the eInk-display, not on the notebook-display.

The different artifacts with black bars that are changing with different LXQT Theme is caused by Openbox?

**Comment by tsujan** [comment]
> ... is caused by Openbox?

Years ago I saw such artifacts with Openbox and without compositor. So, yes, Openbox can create them without a compositor, although some compositors could create artifacts of other kinds if they have bugs.

**Comment by ralf3u** [comment]
The black bars appear in the most cases when the computer is getting out of the suspend-mode.

So, after rebooting the computer in the most cases the bars are not there, but if I go in suspend and then, if I get out of suspend, then the bars are there (but not on the notebook-display, but only on the eInk-display).
Is it also because of Openbox?

**Comment by tsujan** [comment]
How your screen is rendered depends on various factors, among which the most important are the graphic driver, compositor and WM. LXQt is not among them.

**Comment by ralf3u** [comment]
Sorry to ask, but could it be that the `Widget Style` the `Qt Style` "Cleanlooks" is the problem? I ask this because I think that the issue appeared since I installed it.

**Comment by tsujan** [comment]
The Cleanlooks Panel theme is just a stylesheet. It can't cause artifacts. At most, it may trigger the circumstances needed by the real cause.

You'd also mentioned problems about your eInk-display elsewhere (at FeatherPad). My guess is a faulty driver, but your best chance to find the cause is using forums related to that hardware.

**Comment by ralf3u** [comment]
>You'd also mentioned problems about your eInk-display elsewhere (at FeatherPad). My guess is a faulty driver, but your best chance to find the cause is using forums related to that hardware.

No, this is not a problem. I saw the Line Numbers in black background what is normal. I reported the issue at FeatherPad because the idea was to offer an optimized style for eInk-displays, because on eInk-displays it is better when the letters and numbers are in black instead of having black background.

>The Cleanlooks Panel theme is just a stylesheet. It can't cause artifacts.

After the installation of Cleanlooks I turned back to Breeze, and then, I think that even Breeze was strange. I will try to reinstall Lubuntu on an other computer for testing.
Where can I find more stylesheets (Qt Style) that are similar to Cleanlooks to avoid the bars?

Thank you in advance for an answer.
